src/FakeItEasy/Core/DefaultFakeObjectCallFormatter.cs

```diff
@@ -132,15 +132,14 @@ private static void AppendArgumentSeparator(StringBuilder builder, int argumentI
 
         private static ArgumentValueInfo[] GetArgumentValueInfos(IFakeObjectCall call)
         {
-            return
-                (from argument in
-                    call.Method.GetParameters()
-                        .Zip(call.Arguments, (parameter, value) => new { parameter.Name, Value = value })
-                    select new ArgumentValueInfo
-                    {
-                        ArgumentName = argument.Name,
-                        ArgumentValue = argument.Value
-                    }).ToArray();
+            return call.Method.GetParameters()
+                .Zip(call.Arguments, (parameter, value) => new { parameter.Name, Value = value })
+                .Select((argument, index) => new ArgumentValueInfo
+                {
+                    ArgumentIndex = index,
+                    ArgumentName = argument.Name,
+                    ArgumentValue = argument.Value
+                }).ToArray();

```

Comment:
I think I like it better like this anyway... the version using query syntax seemed weird with the parentheses around it.